Yoast/wordpress-seo
# Add basic wpseoApi to query REST Api endpoints.

## Summary

This PR can be summarized in the following changelog entry:

* Registers a `yoast-seo-api` script that can be added as a dependency to expose a `wpseoApi` object on the window that can be used to send requests to Yoast API endpoints.

## Relevant technical choices:

* The API is implemented as the simplest possible object.

## Test instructions

This PR can be tested by following these steps:

* Enqueue the `api` script using the `WPSEO_Admin_Asset_Manager`.
* Open the console and call:
`wpseoApi.get( 'name_of_endpoint', function ( response ) { console.log( result ); } )`
* See if the expected result is logged.

Fixes #7838.
